packages/@aws-cdk/aws-batch/lib/job-queue.ts

```diff
@@ -106,24 +105,16 @@ export class JobQueue extends Resource implements IJobQueue {
   public readonly jobQueueArn: string;
   public readonly jobQueueName: string;
 
-  constructor(scope: Construct, id: string, props: JobQueueProps = {}) {
+  constructor(scope: Construct, id: string, props: JobQueueProps) {
     super(scope, id, {
       physicalName: props.jobQueueName,
     });
 
     const jobQueue = new CfnJobQueue(this, 'Resource', {
-      computeEnvironmentOrder: props.computeEnvironments
-        ? props.computeEnvironments.map(cp => ({
+      computeEnvironmentOrder: props.computeEnvironments.map(cp => ({
           computeEnvironment: cp.computeEnvironment.computeEnvironmentArn,
           order: cp.order,
-        } as CfnJobQueue.ComputeEnvironmentOrderProperty))
-        : [
-            {
-              // Get an AWS Managed Compute Environment
-              computeEnvironment: new ComputeEnvironment(this, 'Resource-Batch-Compute-Environment').computeEnvironmentArn,
-              order: 1,
-            },
-          ],
+        } as CfnJobQueue.ComputeEnvironmentOrderProperty)),
```

Comment:
Can we have a validation on the length (not-zero) of the `computeEnvironment` prop?

Reply:
Yep!